### ISSUE TITLE
cURL checks not working, using TLS1.2

### ISSUE THREAD

**Issue by jpSimkins** [AuthorizeNet/sdk-php#252]
I ran the test with:
```
$ch = curl_init('https://apitest.authorize.net/xml/v1/request.api');
```
and got nothing back. So I followed along and tried:
```
$ch = curl_init('https://www.howsmyssl.com/a/check');
```
and I get 
```
Connection uses: TLS 1.2
```

I am still unable to get the first test working. This leads me to assume that the API will not work for me. 

My cURL is:
```
# curl --version
curl 7.54.1 (x86_64-redhat-linux-gnu) libcurl/7.54.1 OpenSSL/1.0.1e zlib/1.2.3 c-ares/1.13.0 libssh2/1.8.0 nghttp2/1.6.0
Release-Date: 2017-06-14
Protocols: dict file ftp ftps gopher http https imap imaps ldap ldaps pop3 pop3s rtsp scp sftp smb smbs smtp smtps telnet tftp 
Features: AsynchDNS IPv6 Largefile GSS-API Kerberos SPNEGO NTLM NTLM_WB SSL libz HTTP2 UnixSockets HTTPS-proxy Metalink 
```

Running `curl_version` in PHP gives me:
```
array (size=9)
  'version_number' => int 472577
  'age' => int 3
  'features' => int 3113885
  'ssl_version_number' => int 0
  'version' => string '7.54.1' (length=6)
  'host' => string 'x86_64-redhat-linux-gnu' (length=23)
  'ssl_version' => string 'OpenSSL/1.0.1e' (length=14)
  'libz_version' => string '1.2.3' (length=5)
  'protocols' => 
    array (size=22)
      0 => string 'dict' (length=4)
      1 => string 'file' (length=4)
      2 => string 'ftp' (length=3)
      3 => string 'ftps' (length=4)
      4 => string 'gopher' (length=6)
      5 => string 'http' (length=4)
      6 => string 'https' (length=5)
      7 => string 'imap' (length=4)
      8 => string 'imaps' (length=5)
      9 => string 'ldap' (length=4)
      10 => string 'ldaps' (length=5)
      11 => string 'pop3' (length=4)
      12 => string 'pop3s' (length=5)
      13 => string 'rtsp' (length=4)
      14 => string 'scp' (length=3)
      15 => string 'sftp' (length=4)
      16 => string 'smb' (length=3)
      17 => string 'smbs' (length=4)
      18 => string 'smtp' (length=4)
      19 => string 'smtps' (length=5)
      20 => string 'telnet' (length=6)
      21 => string 'tftp' (length=4)
```

This leads me to believe I am using the correct version of cURL.

Doing a `var_dump` on `https://www.howsmyssl.com/a/check`:
```
object(stdClass)[1]
  public 'given_cipher_suites' => 
    array (size=60)
      0 => string 'TLS_ECDHE_RSA_WITH_AES_256_GCM_SHA384' (length=37)
      1 => string 'TLS_ECDHE_ECDSA_WITH_AES_256_GCM_SHA384' (length=39)
      2 => string 'TLS_ECDHE_RSA_WITH_AES_256_CBC_SHA384' (length=37)
      3 => string 'TLS_ECDHE_ECDSA_WITH_AES_256_CBC_SHA384' (length=39)
      4 => string 'TLS_ECDHE_RSA_WITH_AES_256_CBC_SHA' (length=34)
      5 => string 'TLS_ECDHE_ECDSA_WITH_AES_256_CBC_SHA' (length=36)
      6 => string 'TLS_DHE_DSS_WITH_AES_256_GCM_SHA384' (length=35)
      7 => string 'TLS_DHE_RSA_WITH_AES_256_GCM_SHA384' (length=35)
      8 => string 'TLS_DHE_RSA_WITH_AES_256_CBC_SHA256' (length=35)
      9 => string 'TLS_DHE_DSS_WITH_AES_256_CBC_SHA256' (length=35)
      10 => string 'TLS_DHE_RSA_WITH_AES_256_CBC_SHA' (length=32)
      11 => string 'TLS_DHE_DSS_WITH_AES_256_CBC_SHA' (length=32)
      12 => string 'TLS_DHE_RSA_WITH_CAMELLIA_256_CBC_SHA' (length=37)
      13 => string 'TLS_DHE_DSS_WITH_CAMELLIA_256_CBC_SHA' (length=37)
      14 => string 'TLS_ECDH_RSA_WITH_AES_256_GCM_SHA384' (length=36)
      15 => string 'TLS_ECDH_ECDSA_WITH_AES_256_GCM_SHA384' (length=38)
      16 => string 'TLS_ECDH_RSA_WITH_AES_256_CBC_SHA384' (length=36)
      17 => string 'TLS_ECDH_ECDSA_WITH_AES_256_CBC_SHA384' (length=38)
      18 => string 'TLS_ECDH_RSA_WITH_AES_256_CBC_SHA' (length=33)
      19 => string 'TLS_ECDH_ECDSA_WITH_AES_256_CBC_SHA' (length=35)
      20 => string 'TLS_RSA_WITH_AES_256_GCM_SHA384' (length=31)
      21 => string 'TLS_RSA_WITH_AES_256_CBC_SHA256' (length=31)
      22 => string 'TLS_RSA_WITH_AES_256_CBC_SHA' (length=28)
      23 => string 'TLS_RSA_WITH_CAMELLIA_256_CBC_SHA' (length=33)
      24 => string 'TLS_ECDHE_RSA_WITH_AES_128_GCM_SHA256' (length=37)
      25 => string 'TLS_ECDHE_ECDSA_WITH_AES_128_GCM_SHA256' (length=39)
      26 => string 'TLS_ECDHE_RSA_WITH_AES_128_CBC_SHA256' (length=37)
      27 => string 'TLS_ECDHE_ECDSA_WITH_AES_128_CBC_SHA256' (length=39)
      28 => string 'TLS_ECDHE_RSA_WITH_AES_128_CBC_SHA' (length=34)
      29 => string 'TLS_ECDHE_ECDSA_WITH_AES_128_CBC_SHA' (length=36)
      30 => string 'TLS_DHE_DSS_WITH_AES_128_GCM_SHA256' (length=35)
      31 => string 'TLS_DHE_RSA_WITH_AES_128_GCM_SHA256' (length=35)
      32 => string 'TLS_DHE_RSA_WITH_AES_128_CBC_SHA256' (length=35)
      33 => string 'TLS_DHE_DSS_WITH_AES_128_CBC_SHA256' (length=35)
      34 => string 'TLS_DHE_RSA_WITH_AES_128_CBC_SHA' (length=32)
      35 => string 'TLS_DHE_DSS_WITH_AES_128_CBC_SHA' (length=32)
      36 => string 'TLS_DHE_RSA_WITH_SEED_CBC_SHA' (length=29)
      37 => string 'TLS_DHE_DSS_WITH_SEED_CBC_SHA' (length=29)
      38 => string 'TLS_DHE_RSA_WITH_CAMELLIA_128_CBC_SHA' (length=37)
      39 => string 'TLS_DHE_DSS_WITH_CAMELLIA_128_CBC_SHA' (length=37)
      40 => string 'TLS_ECDH_RSA_WITH_AES_128_GCM_SHA256' (length=36)
      41 => string 'TLS_ECDH_ECDSA_WITH_AES_128_GCM_SHA256' (length=38)
      42 => string 'TLS_ECDH_RSA_WITH_AES_128_CBC_SHA256' (length=36)
      43 => string 'TLS_ECDH_ECDSA_WITH_AES_128_CBC_SHA256' (length=38)
      44 => string 'TLS_ECDH_RSA_WITH_AES_128_CBC_SHA' (length=33)
      45 => string 'TLS_ECDH_ECDSA_WITH_AES_128_CBC_SHA' (length=35)
      46 => string 'TLS_RSA_WITH_AES_128_GCM_SHA256' (length=31)
      47 => string 'TLS_RSA_WITH_AES_128_CBC_SHA256' (length=31)
      48 => string 'TLS_RSA_WITH_AES_128_CBC_SHA' (length=28)
      49 => string 'TLS_RSA_WITH_SEED_CBC_SHA' (length=25)
      50 => string 'TLS_RSA_WITH_CAMELLIA_128_CBC_SHA' (length=33)
      51 => string 'TLS_ECDHE_RSA_WITH_3DES_EDE_CBC_SHA' (length=35)
      52 => string 'TLS_ECDHE_ECDSA_WITH_3DES_EDE_CBC_SHA' (length=37)
      53 => string 'TLS_DHE_RSA_WITH_3DES_EDE_CBC_SHA' (length=33)
      54 => string 'TLS_DHE_DSS_WITH_3DES_EDE_CBC_SHA' (length=33)
      55 => string 'TLS_ECDH_RSA_WITH_3DES_EDE_CBC_SHA' (length=34)
      56 => string 'TLS_ECDH_ECDSA_WITH_3DES_EDE_CBC_SHA' (length=36)
      57 => string 'TLS_RSA_WITH_3DES_EDE_CBC_SHA' (length=29)
      58 => string 'TLS_RSA_WITH_IDEA_CBC_SHA' (length=25)
      59 => string 'TLS_EMPTY_RENEGOTIATION_INFO_SCSV' (length=33)
  public 'ephemeral_keys_supported' => boolean true
  public 'session_ticket_supported' => boolean false
  public 'tls_compression_supported' => boolean false
  public 'unknown_cipher_suite_supported' => boolean false
  public 'beast_vuln' => boolean false
  public 'able_to_detect_n_minus_one_splitting' => boolean false
  public 'insecure_cipher_suites' => 
    object(stdClass)[2]
      public 'TLS_DHE_DSS_WITH_3DES_EDE_CBC_SHA' => 
        array (size=1)
          0 => string 'uses 3DES which is vulnerable to the Sweet32 attack but was not configured as a fallback in the ciphersuite order' (length=113)
      public 'TLS_DHE_RSA_WITH_3DES_EDE_CBC_SHA' => 
        array (size=1)
          0 => string 'uses 3DES which is vulnerable to the Sweet32 attack but was not configured as a fallback in the ciphersuite order' (length=113)
      public 'TLS_ECDHE_ECDSA_WITH_3DES_EDE_CBC_SHA' => 
        array (size=1)
          0 => string 'uses 3DES which is vulnerable to the Sweet32 attack but was not configured as a fallback in the ciphersuite order' (length=113)
      public 'TLS_ECDHE_RSA_WITH_3DES_EDE_CBC_SHA' => 
        array (size=1)
          0 => string 'uses 3DES which is vulnerable to the Sweet32 attack but was not configured as a fallback in the ciphersuite order' (length=113)
      public 'TLS_ECDH_ECDSA_WITH_3DES_EDE_CBC_SHA' => 
        array (size=1)
          0 => string 'uses 3DES which is vulnerable to the Sweet32 attack but was not configured as a fallback in the ciphersuite order' (length=113)
      public 'TLS_ECDH_RSA_WITH_3DES_EDE_CBC_SHA' => 
        array (size=1)
          0 => string 'uses 3DES which is vulnerable to the Sweet32 attack but was not configured as a fallback in the ciphersuite order' (length=113)
      public 'TLS_RSA_WITH_3DES_EDE_CBC_SHA' => 
        array (size=1)
          0 => string 'uses 3DES which is vulnerable to the Sweet32 attack but was not configured as a fallback in the ciphersuite order' (length=113)
  public 'tls_version' => string 'TLS 1.2' (length=7)
  public 'rating' => string 'Bad' (length=3)
```

I have tried to prevent `3DES` but no luck. 

At this point, I really don't know what the next step would be to get this to work. 
Any Ideas?

**Comment by NexWeb** [comment]
**!3DES** For Apache (presuming your Apache config is in /etc/httpd/):
grep -r "SSLCipherSuite" /etc/httpd/

For Nginx (presuming your config files are in /etc/nginx/):
grep -r "ssl_ciphers" /etc/nginx/

Once you’ve found the file containing your cipher suite, make sure it contains ‘!3DES’.

**Comment by jpSimkins** [comment]
Ah, sorry. I am using Apache on CentOS 6. I did already have `!3DES` added to my cypher suite. I have since replaced with your example and I still get the same error as I posted in the original post. 

When running your code, I get:
```
123456ErrorE00007User authentication failed due to invalid authentication values.
```

I still get the exact same response from: `https://www.howsmyssl.com/a/check`

I decided to skip the checks part and try the API. The API works fine so I am just going to ignore these check issues.
